src/lib/data/github.ts

```diff
@@ -32,25 +33,130 @@ export class GitHubCollector {
     }
 
     try {
+      // Fetch basic repo data
       const response = await fetch(`https://api.github.com/repos/${repoInfo.owner}/${repoInfo.repo}`);
       if (!response.ok) {
         throw new Error(`GitHub API returned ${response.status}`);
       }
 
-      const fullResponse = await response.json() as any;
-      this.starCount = fullResponse.stargazers_count || 0;
+      const repoData = await response.json() as any;
+
+      // Fetch all file/directory existence data
+      const repoContents = await this.fetchAllRepoContents(repoInfo.owner, repoInfo.repo);
+
+      // Fetch README content using the discovered files
+      const readmeContent = await this.fetchReadmeContent(repoInfo.owner, repoInfo.repo, repoContents);
+
+      this.rawData = {
+        repoData,
+        repoContents,
+        readmeContent,
+      };
     } catch (error) {
       throw new Error(`GitHub fetch failed: ${error}`);
     }
   }
 
-  getStarCount(): number {
-    return this.starCount || 0;
+  private async fetchAllRepoContents(owner: string, repo: string): Promise<Record<string, boolean>> {
+    const results: Record<string, boolean> = {};
+    const visitedPaths = new Set<string>();
+
+    try {
+      // Start recursive fetch from root
+      await this.fetchDirectoryContentsRecursive(owner, repo, '', results, visitedPaths);
+    } catch (error) {
+      console.warn(`Failed to fetch repository contents: ${error}`);
+    }
+
+    return results;
   }
 
-  getData(): GitHubData {
-    return {
-      stars: this.getStarCount(),
-    };
+  private async fetchDirectoryContentsRecursive(
```

Comment:
this seems like overkill, no? it stands to reason that your documentation should be easily found. that means either the file lives at the top level, or there is a top-level folder that makes it clear that the file will live there. feels like you don't need to recurse into the entire directory (this may take unwanted time) and instead just look at the top level for api, docs, documentation keywords.